src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -443,6 +461,12 @@ struct TreeTraversal<BVH, Predicates, Callback,
   KOKKOS_FUNCTION void operator()(int queryIndex) const
   {
     auto const &predicate = Access::get(_predicates, queryIndex);
+    search(predicate);
+  }
+
+  template <typename Query>
+  KOKKOS_FUNCTION void search(Query const &predicate) const
+  {
```

Comment:
This code is different from the spatial one. Is there a reason they should be?

Reply:
I just removed the overloads for this variant for now.

Reply:
Why? Would be nice to have both spatial and nearest single-thread variants. Given that for example XGC asked us to find intersection, and if there's no intersection, find the closest.

Reply:
That's what I thought initially. I still added the implementation for `OrderedSpatial` and a corresponding test.
`Nearest` isn't quite clear to me, though, so I would omit it for now.